lzy/allocator/src/main/java/ai/lzy/allocator/services/KuberNodeService.java

```diff
@@ -0,0 +1,145 @@
+package ai.lzy.allocator.services;
+
+import ai.lzy.allocator.alloc.AllocatorMetrics;
+import ai.lzy.allocator.alloc.impl.kuber.NodeController;
+import com.google.common.net.HostAndPort;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.HttpStatus;
+import io.micronaut.http.MediaType;
+import io.micronaut.http.annotation.Controller;
+import io.micronaut.http.annotation.Post;
+import io.micronaut.http.server.util.HttpClientAddressResolver;
+import jakarta.inject.Inject;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.URI;
+import java.net.UnknownHostException;
+import java.net.http.HttpClient;
+import java.time.Duration;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+import static ai.lzy.allocator.alloc.impl.kuber.KuberLabels.NODE_READINESS_LABEL;
+
+@Controller("/kuber_node")
+public class KuberNodeService {
+    private static final Logger LOG = LogManager.getLogger(KuberNodeService.class);
+
+    private static final int NODE_DAEMONSET_PORT = 8042;
+
+    @Inject
+    NodeController nodeController;
+
+    @Inject
+    AllocatorMetrics metrics;
+
+    @Inject
+    HttpClientAddressResolver clientAddressResolver;
+
+    Set<String> readyNodes = new HashSet<>();
+
+    HttpClient httpClient = HttpClient.newBuilder()
+        .connectTimeout(Duration.ofMillis(1000))
+        .followRedirects(HttpClient.Redirect.NEVER)
+            .build();
+
+    @Post(value = "/set_ready", consumes = MediaType.APPLICATION_JSON)
+    public HttpResponse<String> status(HttpRequest<Map<String, String>> request) {
+        var requestBody = request.getBody().orElse(Map.of());
+        final String clientAddress = clientAddressResolver.resolve(request);
+        final String clusterId = requestBody.get("cluster_id");
+        final String nodeName = requestBody.get("node_name");
+
+        if (clusterId == null || nodeName == null) {
+            String errorMessage = "Required data is not specified";
+            handleError(clientAddress, clusterId, nodeName, errorMessage);
+            return HttpResponse.status(HttpStatus.BAD_REQUEST);
+        }
+
+        LOG.info("Set node ready request; clientAddress: {}, clusterId: {}, nodeName: {}",
+            clientAddress, clusterId, nodeName);
+
+        if (readyNodes.contains(nodeName)) {
+            String errorMessage = "Node was set ready before";
+            handleError(clientAddress, clusterId, nodeName, errorMessage);
+            return HttpResponse.status(HttpStatus.BAD_REQUEST);
```

Comment:
I suppose `HttpStatus.CONFLICT` is more suitable here

Reply:
I don't think we should reply different statuses on errors. It isn't necessary for daemonset, and is definitely not needed for another unexpected client.